firmware/src/main.rs

```diff
@@ -67,55 +73,74 @@ fn main() -> ! {
 }
 
 #[task]
-async fn control_task(packets: &'static Packets, quat: &'static Unreliable<Quat>) {
+async fn control_task(packets: &'static Packets, quat: &'static Unreliable<Quat>) -> ! {
 	debug!("Control task!");
 
-	loop {
-		match packets.clientbound.recv().await {
-			// Identify ourself when discovery packet is received
-			CbPacket::Discovery => {
-				packets
-					.serverbound
-					.send(SbPacket::Handshake {
-						board: 4,
-						imu: 8,
-						mcu_type: 2,
-						imu_info: (1, 2, 3),
-						build: 5,
-						firmware: "SlimeVR-Rust".into(),
-						mac_address: [0; 6],
-					})
-					.await;
-			}
-			// When heartbeat is received, we should reply with heartbeat 0 aka Discovery
-			// The protocol is asymmetric so its a bit unintuitive. TODO: Split packet type into Server2Client and C2S
-			CbPacket::Heartbeat => {
-				packets.serverbound.send(SbPacket::Heartbeat).await;
-			}
-			// Pings are basically like heartbeats, just echo data back
-			CbPacket::Ping { id } => {
-				packets.serverbound.send(SbPacket::Ping { id }).await;
-			}
+	let sensor_info_fut = async {
```

Comment:
this should only happen once at startup

Reply:
at startup, or after the Discovery packet?

Reply:
After discovery packet, or when you get an IMU error